python/mlc_llm/testing/debug_engine.py

```diff
@@ -71,196 +32,8 @@ def _create_tvm_module(
-            hidden_size = head_dim * num_qo_heads
-        kv_bytes_per_token += head_dim * num_kv_heads * num_layers * 4 + 1.25
-        kv_aux_workspace_bytes += (
-            (max_num_sequence + 1) * 88
-            + prefill_chunk_size * (num_qo_heads + 1) * 8
-            + prefill_chunk_size * head_dim * (num_qo_heads + num_kv_heads) * 4
-            + 48 * 1024 * 1024
-        )
-        model_workspace_bytes += (
-            prefill_chunk_size * 4
-            + max_num_sequence * 4
-            + (prefill_chunk_size * 2 + max_num_sequence) * hidden_size * 2
-        )
-        logit_processor_workspace_bytes += (
-            max_num_sequence * 20 + max_num_sequence * vocab_size * 16.125
-        )
-
-    # Get single-card GPU size.
-    gpu_size_bytes = os.environ.get("MLC_GPU_SIZE_BYTES", default=None)
-    if gpu_size_bytes is None:
-        gpu_size_bytes = models[0].device.total_global_memory
-        if gpu_size_bytes is None:
-            raise ValueError(
-                "Cannot read total GPU global memory from device. "
-                'Please the GPU memory size in bytes through "MLC_GPU_SIZE_BYTES" env variable.'
-            )
-
-    max_total_sequence_length = int(
-        (
-            int(gpu_size_bytes) * 0.90
-            - params_bytes
-            - temp_func_bytes
-            - kv_aux_workspace_bytes
-            - model_workspace_bytes
-            - logit_processor_workspace_bytes
-        )
-        / kv_bytes_per_token
-    )
-    assert max_total_sequence_length > 0, (
-        "Cannot estimate KV cache capacity. "
-        f"The model weight size {params_bytes} may be larger than GPU memory size {gpu_size_bytes}"
-    )
-
-    if models[0].device.device_type == Device.kDLMetal:
-        # NOTE: Metal runtime has severe performance issues with large buffers.
-        # To work around the issue, we limit the KV cache capacity to 32768.
-        max_total_sequence_length = min(max_total_sequence_length, 32768)
-
-    total_size = (
-        params_bytes
-        + temp_func_bytes
-        + kv_aux_workspace_bytes
-        + model_workspace_bytes
-        + logit_processor_workspace_bytes
-        + kv_bytes_per_token * max_total_sequence_length
-    )
-    logger.info(
-        "%s: %d.",
-        green('Estimated KVCacheConfig "max_total_sequence_length"'),
-        max_total_sequence_length,
-    )
-    logger.info(
-        "%s: %.2f MB (Parameters: %.2f MB. KVCache: %.2f MB. Temporary buffer: %.2f MB)",
-        green("Estimated total single GPU memory usage"),
-        total_size / 1024 / 1024,
-        params_bytes / 1024 / 1024,
-        (kv_bytes_per_token * max_total_sequence_length + kv_aux_workspace_bytes) / 1024 / 1024,
-        (model_workspace_bytes + logit_processor_workspace_bytes + temp_func_bytes) / 1024 / 1024,
-    )
-    return int(max_total_sequence_length)
-
-
-class Engine:
-    """The Python interface of request serving engine for MLC LLM.
+class DebugEngine:
```

Comment:
DebugSyncEngine

Reply:
Add a quick note about what is the difference from the existing engine impl. This engine wraps Engine without threading so and can be run in sync fashion